core/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java

```diff
@@ -202,4 +221,29 @@ public int hashCode() {
         result = 31 * result + (unit != null ? unit.hashCode() : 0);
         return result;
     }
+
+    @Override
+    public int compareTo(ByteSizeValue other) {
+        long unitValue = unit.toBytes(1);
+        long otherUnitValue = other.unit.toBytes(1);
+
+        ByteSizeUnit minUnit = unitValue < otherUnitValue ? unit : other.unit;
+        long thisValue = this.convert(minUnit);
+        long otherValue = other.convert(minUnit);
+
+        if (thisValue == otherValue) {
```

Comment:
Finally getting back to this! On this look I noticed that this would make hashcode and equals inconsistent which is yet another thing to fix. I haven't really thought through the right way to make them consistent, but there ought to be a way!